### DISCUSSION TITLE
How can I check if versioned migrations are up to date with the schema?

### DISCUSSION THREAD

**Discussion by LiquidLemon** [ariga/atlas#2636] [Q&A] [answered]
I have an old Python app which uses Peewee ORM for database acccess. I figured out how to configure it to produce migration files with a simple `atlas migrate diff --env=local` but I don't know how to validate that my migration files are up to date with the actual schema definitions. I want to set up my CI so that it will check if somebody forgot to run `atlas migrate diff`. I thought either `validate` or `lint` would do it but that doesn't seem to be the case (in my understanding `validate` is not meant for this and `lint` produces no output). Am I missing something? Basically I want something like `diff --dry-run` or `diff --check` where it would produce an error if a new file would be created.

Here's my config: 

```hcl
data "external_schema" "peewee" {
  program = [
    "helpers/db_schema.sh"
  ]
}

env "local" {
  src = data.external_schema.peewee.url
  dev = "sqlite://dev?mode=memory"
  url = "sqlite://db.sqlite"

  migration {
    dir = "file://migrations"
  }

  format {
    migrate {
      diff = format("{{ sql .  \" \" }}")
    }
  }
}
```

**Comment by ronenlu** [comment]
@LiquidLemon  you can run `atlas migrate diff --env=local`  inside your CI pipeline with the `atlas.hcl` you created, and then check that there is no git diff by this command: `status=$(git status --porcelain)`.

If somebody forgot to run `atlas migrate diff` the state of the `peewee` schema will not match the state of your migration directory and new migration file will be created. If the 2 schemas match, no new migration file will be created.

**Comment by a8m** [comment, marked as answer]
Note, you can also use `schema diff` (with custom `--format`) for this and fail if there are changes:
```bash
atlas schema diff \
  --from "file://migrations" \
  --to env://orm \
  --env local \
  --format '{{ if .Changes }}fail{{ end }}' | grep '.' && exit 1
```
(Or, you can convert it to `if [ -n "$output" ]; then .. fi`).

See the doc for more info: https://atlasgo.io/declarative/diff#compare-external-schemas
